src/components/OfflineWithFeedback.js

```diff
@@ -0,0 +1,125 @@
+import {withNetwork} from './OnyxProvider';
+import networkPropTypes from './networkPropTypes';
+import Text from './Text';
+import styles from '../styles/styles';
+import Tooltip from './Tooltip';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import * as StyleUtils from '../styles/StyleUtils';
+import colors from '../styles/colors';
+
+/**
+ * This component should be used when we are using the offline pattern B (offline with feedback).
+ * You should enclose any element that should have feedback that the action was taken offline and it will take
+ * care of adding the appropriate styles for pending actions and displaying the dismissible error.
+ */
+
+const propTypes = {
+    /** The type of action that's pending  */
+    pendingAction: PropTypes.oneOf(['add', 'update', 'delete']),
+
+    /** The errors to display  */
+    // eslint-disable-next-line react/forbid-prop-types
+    errors: PropTypes.object,
+
+    /** A function to run when the X button next to the error is clicked */
+    onClose: PropTypes.func.isRequired,
+
+    /** The content that needs offline feedback */
+    children: PropTypes.node.isRequired,
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    /** Additional styles to add after local styles. Applied to Pressable portion of button */
+    style: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.object),
+        PropTypes.object,
+    ]),
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    pendingAction: null,
+    errors: null,
+    style: [],
+};
+
+/**
+ * This method applies the strikethrough to all the children passed recursively
+ * @param {Array} children
+ * @return {Array}
+ */
+function applyStrikeThrough(children) {
+    return React.Children.map(children, (child) => {
+        if (!React.isValidElement(child)) {
+            return child;
+        }
+        const props = {style: StyleUtils.combineStyles(child.props.style, styles.offlineFeedback.deleted)};
+        if (child.props.children) {
+            props.children = applyStrikeThrough(child.props.children);
+        }
+        return React.cloneElement(child, props);
+    });
+}
+
+const OfflineWithFeedback = (props) => {
+    const isOfflinePendingAction = props.network.isOffline && props.pendingAction;
+    const isUpdateOrDeleteError = props.errors && (props.pendingAction === 'delete' || props.pendingAction === 'update');
+    const isAddError = props.errors && props.pendingAction === 'add';
+    const needsOpacity = (isOfflinePendingAction && !isUpdateOrDeleteError) || isAddError;
+    const needsStrikeThrough = props.network.isOffline && props.pendingAction === 'delete';
+    const hideChildren = !props.network.isOffline && props.pendingAction === 'delete' && !props.errors;
+    let children = props.children;
+    const sortedErrors = _.map(_.sortBy(_.keys(props.errors)), key => props.errors[key]);
```

Comment:
```suggestion
    const sortedErrors = _.chain(props.errors)
        .keys()
        .sortBy()
        .map(key => props.errors[key])
        .value();
```

This may be more readable